testing/spectest/shared/altair/light_client/single_merkle_proof.go

```diff
@@ -0,0 +1,99 @@
+package light_client
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/core/helpers"
+	state_native "github.com/OffchainLabs/prysm/v6/beacon-chain/state/state-native"
+	"github.com/OffchainLabs/prysm/v6/container/trie"
+	ethpb "github.com/OffchainLabs/prysm/v6/proto/prysm/v1alpha1"
+	"github.com/OffchainLabs/prysm/v6/testing/require"
+	"github.com/OffchainLabs/prysm/v6/testing/spectest/utils"
+	"github.com/OffchainLabs/prysm/v6/testing/util"
+	"github.com/golang/snappy"
+)
+
+// RunLightClientSingleMerkleProofTests executes "light_client/single_merkle_proof/{BeaconState}" tests.
+func RunLightClientSingleMerkleProofTests(t *testing.T, config string) {
+	require.NoError(t, utils.SetConfig(t, config))
+
+	_, testsFolderPath := utils.TestFolders(t, config, "altair", "light_client/single_merkle_proof")
+	testTypes, err := util.BazelListDirectories(testsFolderPath)
+	require.NoError(t, err)
+
+	if len(testTypes) == 0 {
+		t.Fatalf("No test types found for %s", testsFolderPath)
+	}
+
+	for _, testType := range testTypes {
+		testFolders, testsFolderPath := utils.TestFolders(t, config, "altair", fmt.Sprintf("light_client/single_merkle_proof/%s", testType))
+		for _, folder := range testFolders {
+			helpers.ClearCache()
+			t.Run(fmt.Sprintf("%v/%v", testType, folder.Name()), func(t *testing.T) {
+				folderPath := path.Join(testsFolderPath, folder.Name())
+				runLightClientSingleMerkleProofTest(t, folderPath, folder.Name())
+			})
+		}
+	}
+}
+
+func runLightClientSingleMerkleProofTest(t *testing.T, testFolderPath string, testName string) {
+	ctx := context.Background()
+
+	beaconStateFile, err := util.BazelFileBytes(path.Join(testFolderPath, "object.ssz_snappy"))
+	require.NoError(t, err)
+	beaconStateSSZ, err := snappy.Decode(nil, beaconStateFile)
+	require.NoError(t, err, "Failed to decompress")
+	beaconStateBase := &ethpb.BeaconStateAltair{}
+	require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+	beaconState, err := state_native.InitializeFromProtoAltair(beaconStateBase)
+	require.NoError(t, err)
+	beaconStateRoot, err := beaconState.HashTreeRoot(ctx)
+	require.NoError(t, err)
+	type Proof struct {
+		Leaf      string   `json:"leaf"`
+		LeafIndex uint64   `json:"leaf_index"`
+		Branch    []string `json:"branch"`
+	}
+	proofFile, err := util.BazelFileBytes(path.Join(testFolderPath, "proof.yaml"))
+	require.NoError(t, err)
+	var proof Proof
+	require.NoError(t, utils.UnmarshalYaml(proofFile, &proof))
+	leaf, err := hex.DecodeString(proof.Leaf[2:])
+	if err != nil {
+		fmt.Printf("Error decoding leaf: %v\n", err)
+	}
+	require.NoError(t, err)
```

Comment:
I think you should also check that the leaf is not empty.

```suggestion
	require.NoError(t, err)
	require.NotEqual(t, 0, len(leaf), "expected non-zero length leaf")
```